indexing-service/src/main/java/org/apache/druid/indexing/common/task/CompactionTask.java

```diff
@@ -1522,4 +1529,10 @@ public CompactionTuningConfig withPartitionsSpec(PartitionsSpec partitionsSpec)
       );
     }
   }
+
+  @Override
+  public LookupLoadingSpec getLookupLoadingSpec()
+  {
+    return LookupLoadingSpec.getSpecFromContext(getContext(), LookupLoadingSpec.NONE);
```

Comment:
I wonder if the `index_parallel` supervisor tasks launched by compaction will ever need lookups. I should think not. So, this should probably always return `NONE`.

cc: @cryptoe

Reply:
@kfaraz Based on offline discussion, compaction tasks can technically reference lookups as part of the transformSpec. But that scenario is rarely ever used, so the conclusion was to disable lookup loading by default, while still keeping the option open to be able to override this behavior and load lookups if needed.

Hope that makes sense.

Reply:
Yes, it does. 

My question is not about the sub-tasks that will be launched by compaction, rather the supervisor tasks of type `index_parallel`. These supervisor tasks run in the same process as the `CompactionTask` itself and never do any indexing themselves.

So I don't think they would ever need the lookups.

Reply:
@kfaraz I might be misunderstanding your comment due to lack of context (about compaction and supervisors in general), but `index_parallel` is also used for batch ingestion, which can reference lookups. So I'm not sure I understand why they might never need lookups?

Appreciate if you could elaborate, thanks!